lib/java/kafka/streams/src/main/java/co/airy/kafka/streams/KafkaStreamsWrapper.java

```diff
@@ -174,16 +174,19 @@ public KafkaStreamsWrapper withExcludeInternalTopics(boolean excludeInternalTopi
 
     public <T> ReadOnlyKeyValueStore<String, T> acquireLocalStore(String storeName) {
         int retries = 0;
-        do
+        do {
             try {
+                if (!streams.state().equals(KafkaStreams.State.RUNNING)) {
```

Comment:
Prior to [KAFKA-10271](https://github.com/apache/kafka/commit/ccff7227ce11747c758c0903b34ee1d84119b9f9#) the underlying `QueryableStoreProvider` would expensively lookup stores from all registered `StreamThreadStateStoreProvider` instances. `StreamThreadStateStoreProvider::getStore` throws an exception if the thread is not `RUNNING` so there was an implicit check for that whenever we requested a store.

[KAFKA-10271](https://github.com/apache/kafka/commit/ccff7227ce11747c758c0903b34ee1d84119b9f9#) removed that extra lookup for performance reasons so that it became possible to access stores that would crash when trying to read from them.

Reply:
This also explains why the integration tests were failing, while the runtime was working: The state stores do eventually get ready, but for the tests that were failing they were not ready at the expected time.